File: game/src/main/java/game/world/Layout.java
package game.world;

/**
 * @author ismd
 */
public class Layout {

    private Cell[][] cells;

    Layout(int[][] map) {
        cells = new Cell[map.length][];

        for (int i = 0; i < map.length; i++) {
            cells[i] = new Cell[map[i].length];

            for (int j = 0; j < map[i].length; j++) {
                cells[i][j] = new Cell(CellType.getById(map[i][j]));
            }
        }
    }
}

File: site/application/models/Layout.php
<?php
/**
 * Модель карты
 * @author ismd
 */

class Layout extends Abstract_Layout {

    private $_layout = array();

    public function __construct($options = null) {
        parent::__construct($options);

        $file = file_get_contents('img/world/layouts/' . $this->getId() . '.txt');
        $this->_layout = json_decode($file);
    }

    public function getLayout() {
        return $this->_layout;
    }

    public function getWidth() {
        return count($this->getLayout());
    }

    public function getHeight() {
        return count($this->getLayout()[0]);
    }
}
